com.unity.render-pipelines.universal/Runtime/RendererFeatures/DecalRendererFeature.cs

```diff
@@ -313,24 +313,52 @@ private DecalTechnique GetTechnique(bool isDeferred)
         private bool IsAutomaticDBuffer()
         {
 #if UNITY_EDITOR
-            var selectedBuildTargetGroup = UnityEditor.EditorUserBuildSettings.selectedBuildTargetGroup;
-            if (selectedBuildTargetGroup == UnityEditor.BuildTargetGroup.Standalone)
-                return true;
-            if (selectedBuildTargetGroup == UnityEditor.BuildTargetGroup.GameCoreXboxOne)
-                return true;
-            if (selectedBuildTargetGroup == UnityEditor.BuildTargetGroup.GameCoreXboxSeries)
-                return true;
-            if (selectedBuildTargetGroup == UnityEditor.BuildTargetGroup.PS4)
-                return true;
-            if (selectedBuildTargetGroup == UnityEditor.BuildTargetGroup.PS5)
-                return true;
-            if (selectedBuildTargetGroup == UnityEditor.BuildTargetGroup.WSA)
-                return true;
-            if (selectedBuildTargetGroup == UnityEditor.BuildTargetGroup.Switch)
-                return true;
-            return false;
+            switch (UnityEditor.EditorUserBuildSettings.selectedBuildTargetGroup)
+            {
+                // Standalone
+                case UnityEditor.BuildTargetGroup.Standalone:
+                    return true;
+
+                // WSA
+                case UnityEditor.BuildTargetGroup.WSA:
+                    return true;
+
+                // Console
+                case UnityEditor.BuildTargetGroup.GameCoreXboxOne:
+                case UnityEditor.BuildTargetGroup.GameCoreXboxSeries:
+                case UnityEditor.BuildTargetGroup.PS4:
+                case UnityEditor.BuildTargetGroup.PS5:
+                case UnityEditor.BuildTargetGroup.Switch:
+                    return true;
+
+                default:
+                    return false;
```

Comment:
The way this was designed was to be that we allow DBuffer only on platforms that we specifically approved it, while it seems to me that we should have default enabled and disable DBuffer only on platforms that we know are limited in terms of power. We expected that new generations of platforms comes with more processing power.

My justification is that DBuffer is the higher quality / more complete feature set, so now let's say we have a new generation of platform so now that platform gets less quality until someone adds the platform to this switch case, which seems counterintuitive to me.

Reply:
The other assumption I have is that it seems the less powerful platforms are the smaller set, so this switch case grows less than the opposite.

Reply:
It makes sense, but switch statement will be much bigger